pkg/utils/json.go

```diff
@@ -42,3 +47,51 @@ func removeNaN(hosts []result.Host) []result.Host {
 	}
 	return newHosts
 }
+
+func GetMetics(searchDir string, m *[]metrics.Metrics) error {
+	resultFilePath := path.Join(path.Dir(path.Clean(searchDir)), "result.txt")
+
+	bytes, err := ioutil.ReadFile(resultFilePath)
+	if err != nil {
+		return err
+	}
+
+	r, err := regexp.Compile(`{[^}]*}`)
+	if err != nil {
+		return err
+	}
+
+	// TODO import this from origin; for now it is still a private field
+	marker_name := "cluster_loader_marker"
+	var bm metrics.BaseMetrics
+	for _, jsonBytes := range r.FindAll(bytes, -1) {
+		jsonString := string(jsonBytes)
```

Comment:
Why are you converting to string?

Reply:
To use it in the fmt, eg,
fmt.Printf("cannot unmarshal the line '%v' for BaseMetrics: %v\n", jsonString, err)

Could be used in several places ... so do it once.

Reply:
Done.
Put the string conversion into fmt calls.